crates/chain/src/indexer/spk_txout.rs

```diff
@@ -318,6 +320,74 @@ impl<I: Clone + Ord + core::fmt::Debug> SpkTxOutIndex<I> {
         (sent, received)
     }
 
+    /// Collects the sent and received [`TxOut`]s for `tx` on the script pubkeys in `range`.
+    /// TxOuts are *sent* when a script pubkey in the `range` is on an input and *received* when
+    /// it is on an output. For `sent` to be computed correctly, the index must have already
+    /// scanned the output being spent. Calculating received just uses the [`Transaction`]
+    /// outputs directly, so it will be correct even if it has not been scanned.
+    ///
+    /// Returns a tuple of (sent_txouts, received_txouts).
+    ///
+    /// # Example
+    /// Shows the addresses of the TxOut sent from or received by a Transaction relevant to all spks
+    /// in this index.
+    ///
+    /// ```rust
+    /// # use bdk_chain::spk_txout::SpkTxOutIndex;
+    /// # use bitcoin::{Address, Network, Transaction};
+    /// # use std::str::FromStr;
+    /// #
+    /// # fn example() -> Result<(), Box<dyn std::error::Error>> {
+    /// let mut index = SpkTxOutIndex::<u32>::default();
+    ///
+    /// // ... scan transactions to populate the index ...
+    /// # let tx = Transaction { version: bitcoin::transaction::Version::TWO, lock_time: bitcoin::locktime::absolute::LockTime::ZERO, input: vec![], output: vec![] };
+    ///
+    /// // Get sent and received txouts for a transaction across all tracked addresses
+    /// let (sent_txouts, received_txouts) = index.sent_and_received_txouts(&tx, ..);
+    ///
+    /// // Display addresses and amounts
```

Comment:
@tnull would this information be useful for LDKnode when handling the transactions in WalletEvents ?